aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecorator.kt

```diff
@@ -35,6 +35,7 @@ import software.amazon.smithy.rust.codegen.core.util.getTrait
 import software.amazon.smithy.rust.codegen.core.util.hasEventStreamOperations
 import software.amazon.smithy.rust.codegen.core.util.hasTrait
 import software.amazon.smithy.rust.codegen.core.util.isInputEventStream
+import software.amazon.smithy.rust.codegen.core.util.letIf
 import software.amazon.smithy.rust.codegen.core.util.thenSingletonListOf
 
 class SigV4AuthDecorator : ClientCodegenDecorator {
```

Comment:
maybe we should make `ApplicableCodegenDecorator(filter: ClientCodenContext -> bool)` as an abstract class to unify these patterns

Reply:
We have ServiceSpecificDecorator that's 90% of the way there, so I think I'll refactor that to be more generic.

Reply:
Done in 374b39ff288697b2a8c7c552ed66b1130af5ce4e